### PR TITLE
TSK-960: move for issues

### PR DESCRIPTION
**Pull request by ThetaDR**
# Contribution checklist

## Brief description
https://front.hc.engineering/workbench/platform/tracker/TSK-960
## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: plugins/tracker-resources/src/components/issues/Move.svelte [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  let space: Ref<Project>
+
+  $: _class = currentSpace ? hierarchy.getClass(currentSpace._class).label : undefined
+  let classLabel = ''
+  $: {
+    const doc = docs[0]
+    if (space === undefined) space = doc.space
+    translate(hierarchy.getClass(doc._class).label, {}).then((res) => (label = res.toLocaleLowerCase()))
+  }
+  $: _class && translate(_class, {}).then((res) => (classLabel = res.toLocaleLowerCase()))
```
Comment (BykhovDenis) [inline plugins/tracker-resources/src/components/issues/Move.svelte]: We already know class and class name I think

File: plugins/tracker-resources/src/components/issues/Move.svelte [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+  const moveAll = async () => {
+    docs.forEach(async (doc) => await move(doc))
+  }
+
+  async function getSpace (): Promise<void> {
+    client.findOne(tracker.class.Project, { _id: space }).then((res) => (currentSpace = res))
+  }
+
+  async function invokeValidate (
```
Comment (BykhovDenis) [inline plugins/tracker-resources/src/components/issues/Move.svelte]: We don't have object validator for issue, and I think won't have in observable future

File: plugins/tracker-resources/src/utils.ts [hunk 3]
```diff
@@ -685,6 +689,50 @@ export interface StatusStore {
...
+          await moveIssueToSpace(client, attached, space, {
+            ...extra,
+            rank: calcRank(lastOne, undefined),
+            number: (incResult as any).object.sequence
+          }).catch((err) => console.log('failed to move', name, err))
+        } else await moveIssueToSpace(client, attached, space).catch((err) => console.log('failed to move', name, err))
+      }
+    }
+  }
+  await client.update(doc, {
```
Comment (BykhovDenis) [inline plugins/tracker-resources/src/utils.ts]: Let's do it all in one applyTx

**Review by BykhovDenis — CHANGES_REQUESTED** [review]

File: plugins/tracker-resources/src/utils.ts [hunk 4]
```diff
@@ -685,6 +689,50 @@ export interface StatusStore {
...
+): Promise<void> {
+  const hierarchy = client.getHierarchy()
+  const attributes = hierarchy.getAllAttributes(doc._class)
+  for (const [name, attribute] of attributes) {
+    if (hierarchy.isDerived(attribute.type._class, core.class.Collection)) {
+      const collection = attribute.type as Collection<AttachedDoc>
+      const allAttached = await client.findAll(collection.of, { attachedTo: doc._id })
+      for (const attached of allAttached) {
+        // Do not use extra for childs.
+        if (name === 'subIssues') {
```
Comment (haiodo) [inline plugins/tracker-resources/src/utils.ts]: I would recommend to check for collection type is ref to class derived from Issue, so this code will not depend on text constant. Thanks.

**Review by haiodo — APPROVED** [review]

**Review by BykhovDenis — APPROVED** [review]

**Pull request merged**